src/types.js

```diff
@@ -10,6 +10,67 @@ export type TXOBJ = {
     prevScriptPubKey: string,
   }[],
   outs: { script: string, satoshis: number }[],
+  joinsplit?: {}[], // Does ZencashJS support shielded txn?
+  sc_params?: {
+    vsc_ccout: {
+      epoch_length: number,
+      satoshis: number,
+      address: string,
+      customData: string,
+      wCertVk: string,
+      vFieldElementCertificateFieldConfig: number[],
+      vBitVectorCertificateFieldConfig: number[][],
+      forwardTransferScFee: number,
+      mainchainBackwardTransferScFee: number,
+      mbtrRequestDataLength: number
+    }[],
+    vft_ccout: {
+      scid: string,
+      satoshis: number,
+      address: string
+    }[],
+    vcsw_ccin: {
+      value: number,
+      scId: string,
+      nullifer: string,
+      scProof: string,
+      actCertDataHash: string,
+      ceasingCumScTxCommTree: string,
+      redeemScript: string,
+      pubKeyHash: string
+    }[],
+    vmbtr_out: {
+      scid: string,
+      scFee: string,
+      mcDestinationAddress: string,
+      vScRequestData: string[]
+    }[],
+  },
+}
+
+export type CERTIFICATE_OBJ = {
```

Comment:
I suggest to create a Certificate type that is an extension of the Transaction type and add all the certificate related information inside an additional field (for example **cert**). See **sc_params** used for the Sidechain transactions